app/controllers/schools/users_controller.rb

```diff
@@ -0,0 +1,108 @@
+module Schools
+  class UsersController < ApplicationController
+    layout "school"
+
+    before_action :set_user, except: :index
+
```

Comment:
I made an oversight in my previous comment. This index route still needs to be locked down so that only admins are authorized to access it (without regard for the _resource_)...

Right now, a non-admin trying to access this route will cause a crash, instead of a 404.

Add a regular `authorize` call, passing the current user - you still need to do a check in `index?`:

```rb
def index?
  user&.school_admin.present? && user.school == current_school
end
```